src/fn_call.rs

```diff
@@ -831,3 +1038,89 @@ fn gen_random<'a, 'mir, 'tcx>(
         }
     }
 }
+
+/// A helper method to unwind the stack.
+///
+/// We execute the 'unwind' blocks associated with frame
+/// terminators as we go along (these blocks are responsible
+/// for dropping frame locals in the event of a panic)
+///
+/// When we find our target frame, we write the panic payload
+/// directly into its locals, and jump to it.
+/// After that, panic handling is done - from the perspective
+/// of the caller of '__rust_maybe_catch_panic', the function
+/// has 'returned' normally, after which point Miri excecution
+/// can proceeed normally.
+fn unwind_stack<'a, 'mir, 'tcx>(
+    this: &mut MiriEvalContext<'a, 'mir, 'tcx>,
+    payload_data_ptr: Scalar<Borrow>,
+    payload_vtable_ptr: Scalar<Borrow>
+) -> EvalResult<'tcx> {
+
+    let mut found = false;
+
+    while !this.stack().is_empty() {
+        // When '__rust_maybe_catch_panic' is called, it marks is frame
+        // with 'catch_panic'. When we find this marker, we've found
+        // our target frame to jump to.
+        if let Some(unwind_data) = this.frame_mut().extra.catch_panic.take() {
+            trace!("unwinding: found target frame: {:?}", this.frame().span);
+
+            let data_ptr = unwind_data.data_ptr.clone();
+            let vtable_ptr = unwind_data.vtable_ptr.clone();
+            let dest = unwind_data.dest.clone();
+            let ret = unwind_data.ret.clone();
+            drop(unwind_data);
+
+
+            // Here, we write directly into the frame of the function
+            // that called '__rust_maybe_catch_panic'.
+            // (NOT the function that called '__rust_start_panic')
+
+            this.write_scalar(payload_data_ptr, data_ptr.into())?;
+            this.write_scalar(payload_vtable_ptr, vtable_ptr.into())?;
+
+            // We 'return' the value 1 from __rust_maybe_catch_panic,
+            // since there was a panic
+            this.write_scalar(Scalar::from_int(1, dest.layout.size), dest)?;
+
+            // We're done - continue execution in the frame of the function
+            // that called '__rust_maybe_catch_panic,'
+            this.goto_block(Some(ret))?;
+            found = true;
```

Comment:
instead of setting a bool variable, `return Ok(());` directly from here and make the trailing expression of the function `err!(Unreachable)`